assets/js/src/core/components/tag/tag.tsx

```diff
@@ -38,19 +51,38 @@ export const Tag = ({ children, icon, iconName, theme, className, ...props }: Ta
     />
   )
 
-  const getIcon = (): React.ReactElement<any, string | React.JSXElementConstructor<any>> | string | number | Iterable<React.ReactNode> | React.ReactPortal | null | boolean => {
-    if (iconName !== undefined) return renderIcon(iconName)
-    else if (icon !== undefined) return icon
-    else return null
+  const getIcon = (): React.ReactNode => {
+    if (iconName !== undefined) {
+      return renderIcon(iconName)
+    }
+
+    return icon ?? null
   }
 
-  return (
+  const renderTag = (): JSX.Element => (
     <AntTag
-      className={ classes.join(' ') }
+      className={ tagClassNames }
       icon={ getIcon() }
       { ...props }
     >
-      {children}
+      {displayText}
     </AntTag>
   )
+
+  return (
+    <>
+      {shouldShowTooltip
+        ? (
+          <Tooltip
+            align={ { offset: [20, -4] } }
+            overlayClassName={ styles.tooltip }
+            title={ children }
```

Comment:
I'm not sure if the title should always be the direct children. I also could imagine that there is a longer descriptive text in there, compared to the content. Maybe we add a new Prop tooltip title? 

We also could omit the shouldShowTooltip prop then, and only show the tooltip when a tooltip title is passed as a prop.

Reply:
hmm, but that would also mean, that you can't really control the tooltip based on the maxLength anymore.
I don't know for me it seems like we are limiting us here a bit. Maybe it would be useful to have some kind of truncate utility and just handle the cutting of the content outside of the tag component? 

What do you think?

Reply:
> I'm not sure if the title should always be the direct children. I also could imagine that there is a longer descriptive text in there, compared to the content. Maybe we add a new Prop tooltip title?
> 
> We also could omit the shouldShowTooltip prop then, and only show the tooltip when a tooltip title is passed as a prop.

I see what you mean. However, on the other hand, it could be a bit tiring to pass _tooltipTitle_ each time. That’s why I propose to keep the current behavior as the default and additionally add _tooltipTitle_ prop to cover the case you mentioned earlier. What is your opinion?

Reply:
> hmm, but that would also mean, that you can't really control the tooltip based on the maxLength anymore. I don't know for me it seems like we are limiting us here a bit. Maybe it would be useful to have some kind of truncate utility and just handle the cutting of the content outside of the tag component?
> 
> What do you think?

From my perspective, it’s good to have the opportunity to set a character limit and show the Tooltip depending on the Tag content (we can either set this value or not). Otherwise, we would need to define rules for when to truncate the Tag content, which could lead to additional edge cases that we would need to handle.

Reply:
Lastly, showing a tooltip based on the maximum character length was a requirement from @idaiv . If we want to change it, we need to have an additional grooming session with Ida.